std/conv.d

```diff
@@ -1883,6 +1882,48 @@ if (isInputRange!S && !isInfinite!S && isSomeChar!(ElementEncodingType!S) &&
     assert(to!int(byDchar("10"), 10) == 10);
 }
 
+/**
+String, or string-like input range, to char type not directly
+supported by parse parses the first dchar of the source.
+
+Returns: the first code point of the input range, converted
+         to type T.
+
+Throws: ConvException if the input range contains more than
+        a single code point, or if the code point does not
+        fit into a code unit of type T.
+*/
+private T toImpl(T, S)(S value)
+if (isInputRange!S && isSomeChar!(ElementEncodingType!S) &&
+     isSomeChar!T && !is(typeof(parse!T(value))) &&
+     is(typeof(parse!double(value))))
```

Comment:
This looks weird. There shouldn't be a need to check for `parse!double`

Reply:
I use parse!double(value) within the method, and given the argument is of type S, it may not compile. I'll check and see if instantiation conditions for parse overlap the ones on S for this method,